bentoml/_internal/frameworks/transformers.py

```diff
@@ -106,427 +65,209 @@ def check_flax_supported() -> None:  # pragma: no cover
-    else:
-        config_kwargs = kwargs.pop("config_kwargs", {})
-        config: "ext.PretrainedConfig" = AutoConfig.from_pretrained(
-            model.path, **config_kwargs
-        )
 
-        model_kwargs = kwargs.pop("model_kwargs", {})
+def load_model(
+    bento_model: str | Tag | Model,
+    **kwargs: t.Any,
+) -> "ext.TransformersPipeline":
+    """
+    Load the Transformers model from BentoML local modelstore with given name.
 
-        _model, _tokenizer = (
-            model.info.options["model"],
-            model.info.options["tokenizer"],
-        )
-        _feature_extractor = model.info.options["feature_extractor"]
+    Args:
+        bento_model (``str`` ``|`` :obj:`~bentoml.Tag` ``|`` :obj:`~bentoml.Model`):
+            Either the tag of the model to get from the store, or a BentoML `~bentoml.Model`
+            instance to load the model from.
+        kwargs (:code:`Any`):
+            Additional keyword arguments to pass to the model.
 
-        if _tokenizer is False:
-            tokenizer: t.Optional["ext.TransformersTokenizerType"] = None
-        else:
-            tokenizer_kwargs = kwargs.pop("tokenizer_kwargs", {})
-            tokenizer = getattr(
-                import_module("transformers"), _tokenizer
-            ).from_pretrained(
-                model.path, from_tf=from_tf, from_flax=from_flax, **tokenizer_kwargs
-            )
-        if _feature_extractor is False:
-            feature_extractor: t.Optional["ext.PreTrainedFeatureExtractor"] = None
-        else:
-            feature_extractor_kwargs = kwargs.pop("feature_extractor_kwargs ", {})
-            feature_extractor = getattr(
-                import_module("transformers"), _feature_extractor
-            ).from_pretrained(model.path, **feature_extractor_kwargs)
+    Returns:
+        ``Pipeline``:
+            The Transformers pipeline loaded from the model store.
 
-        tfe = tokenizer if tokenizer is not None else feature_extractor
+    Example:
+    .. code-block:: python
+        import bentoml
+        pipeline = bentoml.transformers.load_model('my_model:latest')
+    """  # noqa
+    if not isinstance(bento_model, Model):
+        bento_model = get(bento_model)
 
-        tmodel: "ext.TransformersModelType" = getattr(import_module("transformers"), _model).from_pretrained(  # type: ignore[reportUnknownMemberType]
-            model.path,
-            config=config,
-            **model_kwargs,
+    if bento_model.info.module not in (MODULE_NAME, __name__):
+        raise BentoMLException(
+            f"Model {bento_model.tag} was saved with module {bento_model.info.module}, failed loading with {MODULE_NAME}."
         )
 
-        if return_config:
-            return config, tmodel, tfe  # type: ignore
-        return None, tmodel, tfe  # type: ignore
+    return transformers.pipeline(
+        bento_model.info.options["task"], bento_model.path, **kwargs
+    )
```

Comment:
lets also do 
```python
            try:
                transformers.pipelines.check_task(task)  # type: ignore
            except KeyError as e:
                raise BentoMLException(
                    f"{e}, as `{task}` is not recognized by transformers."
```

Reply:
Good catch. I missed this check.

Reply:
Added as a part of the `attrs` class field validator.

Reply:
noted